pyproject.toml

```diff
@@ -18,12 +18,19 @@ requires-python = ">=3.11"
 
 dependencies = [
   "boto3",
+  "numpy",
```

Comment:
Tempted to suggest we make this an optional dependency as well, since all the S3 utils don't need it.